pilot/pkg/features/experimental.go

```diff
@@ -217,9 +217,25 @@ var (
 	EnableOptimizedServicePush = env.RegisterBoolVar("ISTIO_ENABLE_OPTIMIZED_SERVICE_PUSH", true,
 		"If enabled, Istiod will not push changes on arbitrary annotation change.").Get()
 
-	// This is used in injection templates, it is not unused.
-	EnableNativeSidecars = env.Register("ENABLE_NATIVE_SIDECARS", false,
-		"If set, used Kubernetes native Sidecar container support. Requires SidecarContainer feature flag.")
+	EnableNativeSidecars = func() NativeSidecarMode {
+		v := env.Register("ENABLE_NATIVE_SIDECARS", "false",
+			"If set, used Kubernetes native Sidecar container support. Requires SidecarContainer feature flag."+
+				" Set to true or false to unconditionally enable. Set to auto-beta or auto-stable to automatically enable"+
+				" if support is detected (at the beta or stable level).").Get()
+		switch v {
+		case "never", "false":
+			return NativeSidecarModeNever
+		case "always", "true":
+			return NativeSidecarModeAlways
+		case "auto-beta":
```

Comment:
Eventually I think the default should be `auto-stable`. We should not default to beta k8s features, only stable ones IMO. So the possible one to remove is `auto-beta` IMO.

The benefit of auto-beta is that a lot of users _do_ want to use beta k8s features, and its IMO pretty reasonable to. So what they get is a bit of safety in doing that, rather than just `on`.

Most of the complexity is in the detection anyways, beta vs stable is pretty simple

Reply:
Can we just do 

`never`, `always` and `auto` for now?

Whether `auto` should ignore k8s beta features or not probably should be controlled globally by e.g. @whitneygriffith's `experimental/stable` profiles and implicit in that (if you are in `istio stable` you do not get to use `k8s experimental` stuff, and if you are in `istio experimental` you do), and not at the feature level.